cmake/Config.cmake

```diff
@@ -28,7 +47,27 @@ find_package (Boost 1.53.0
         thread
         system
         unit_test_framework
-        python)
+        ${_BOOSTPYLIB})
+
+# if _BOOSTPYLIB was not found for non-windows platforms,
+# try searching for built-from-source defaults
+if (NOT WIN32 AND NOT ${Boost_${_BOOSTPYLIB}_FOUND})
+    if (${PYTHON_VERSION_MAJOR} STREQUAL 2)
+        set (_BOOSTPYLIB "python")
+    else()
+        set (_BOOSTPYLIB "python${PYTHON_VERSION_MAJOR}")
+    endif()
+    find_package (Boost 1.53.0
+        OPTIONAL_COMPONENTS
+            date_time
+            thread
+            system
+            unit_test_framework
+            ${_BOOSTPYLIB})
+endif()
+# remap boost python vars
+set(Boost_PYTHON_FOUND ${Boost_${_BOOSTPYLIB}_FOUND})
+set(Boost_PYTHON_LIBRARY ${Boost_${_BOOSTPYLIB}_FOUND})
```

Comment:
This doesn't work, Python examples/unit tests are not built (tested with cmake 3.0.2 and Python 2.7). 

First of all the second `set` has a clear bug, it should not use `_FOUND` variable. But there are also other subtleties of cmake (`_LIBRARY` variable can have multiple values) and what looks to me like bug in cmake (the first `set` should work but doesn't). Here's a version that seems more robust and I verified it in the above environment:

```
if (Boost_${_BOOSTPYLIB}_FOUND)
    set (Boost_PYTHON_FOUND ON)
    set (Boost_PYTHON_LIBRARY "${Boost_${_BOOSTPYLIB}_LIBRARY}")
endif() 
```

Can you please check if that also works on Linux and Python 3?